### PR TITLE
Created Metric to track whether tablet is leader

### PR DESCRIPTION
**Pull request by Adm28**
#3688 
Created Metric `is_leader` which tracks whether a tablet is leader/follower.
Plugged in the Metric into `raft_consensus.cc` to determine when the leader becomes leader and steps down from leader. 
The metric does so in the following way 

- Set the metric to 1 at the end of  func `RaftConsensus::BecomeLeaderUnlocked()` (Becomes a leader)
- Set the metric to 0 at the end of func `RaftConsensus::BecomeReplicaUnlocked()`(Steps down)

### CODE REVIEW

File: src/yb/consensus/raft_consensus.cc [hunk 1]
```diff
@@ -155,6 +155,12 @@ METRIC_DEFINE_lag(tablet, follower_lag_ms,
                   "The amount of time since the last UpdateConsensus request from the "
                   "leader.");
 
+METRIC_DEFINE_gauge_int32(tablet, is_leader,
```
Comment (rahuldesirazu) [inline src/yb/consensus/raft_consensus.cc:158]: Call this is_raft_leader to make it clear this is a raft level metric.

**Comment by rahuldesirazu** [comment]
And one more thing if you don't mind, can you add your github handle to https://github.com/yugabyte/yugabyte-db/blob/master/CONTRIBUTORS.md. Thanks!

**Comment by Adm28** [comment]
> Great work, thanks for becoming a contributor! I would add a very short unit test that creates a table with 1 tablet in create-table-itest.cc and ensure that there is 1 leader and 2 followers (See GetInt64Metric in external_mini_cluster.h)
Thanks for your review **@rahuldesirazu**. I am sorry but I didn't understand how after creating a table with a single tablet can we expect to have 1 leader and 2 followers. Where are the other two tablets coming into the scene ?

**Comment by Adm28** [comment]
> Great work, thanks for becoming a contributor! I would add a very short unit test that creates a table with 1 tablet in create-table-itest.cc and ensure that there is 1 leader and 2 followers (See GetInt64Metric in external_mini_cluster.h)

**Comment by rahuldesirazu** [comment]
@Adm28 If we create a table with 1 tablet and replication factor of 3, we should expect there to be one leader and two followers by looking at the metric. Does that make sense?

**Comment by Adm28** [comment]
> @Adm28 If we create a table with 1 tablet and replication factor of 3, we should expect there to be one leader and two followers by looking at the metric. Does that make sense?

Thanks for clarifying @rahuldesirazu , makes sense.

**Comment by Adm28** [comment]
> Great work, thanks for becoming a contributor! I would add a very short unit test that creates a table with 1 tablet in create-table-itest.cc and ensure that there is 1 leader and 2 followers (See GetInt64Metric in external_mini_cluster.h)

Sorry @rahuldesirazu ,I misinterpreted that I had to write the unit test case. Should I discard my unit-test ?

**Comment by Adm28** [comment]
> Great work, thanks for becoming a contributor! I would add a very short unit test that creates a table with 1 tablet in create-table-itest.cc and ensure that there is 1 leader and 2 followers (See GetInt64Metric in external_mini_cluster.h)

I misread this and thought I had to add the unit test. Should I discard the unit test ?

**Comment by rahuldesirazu** [comment]
The unit test you added looks great, thanks for adding it! The code looks good, now going to run through our jenkins and confirm tests pass.

**Comment by Adm28** [comment]
> The unit test you added looks great, thanks for adding it! The code looks good, now going to run through our jenkins and confirm tests pass.

@rahuldesirazu Any updates regarding the tests ?

**Comment by rahuldesirazu** [comment]
@Adm28 The tests all pass. I ended up making a few small changes to satisfy our linter- purely style stuff. Will add a commit with those changes and merge. Thanks again for your contribution!

**Comment by rahuldesirazu** [comment]
@Adm28 One last thing, would you mind cherry-picking the most recent commit on https://github.com/yugabyte/yugabyte-db/pull/5136 with the style fixes- I can't make the commit without creating a new pull request. Thanks!

**Comment by Adm28** [comment]
Thanks @rahuldesirazu for satisfying the linter requirements. I have cherry picked the required commit onto my master. Let me know if you require something else from my side.

**Comment by rahuldesirazu** [comment]
Just merged, thanks again @Adm28!

**Comment by Adm28** [comment]
@rahuldesirazu You can close #3688 
Thanks

**Pull request merged**